modules/phase_field/src/kernels/KKSSplitCHCRes.C

```diff
@@ -111,11 +95,7 @@ KKSSplitCHCRes::computeQpOffDiagJacobian(unsigned int jvar)
   if (jvar == _w_var)
     return -_phi[_j][_qp] * _test[_i][_qp];
 
-  if (jvar == _ca_var)
-    return _phi[_j][_qp] * _test[_i][_qp] * _second_derivative_Fa[_qp];
-
   // get the coupled variable jvar is referring to
   const unsigned int cvar = mapJvarToCvar(jvar);
-
   return _phi[_j][_qp] * _test[_i][_qp] * (*_d2Fadcadarg[cvar])[_qp];
```

Comment:
I'm assuming the `d2Fadcadarg[cvar]` is now going to catch the case `(jvar == _ca_var)` that has been removed. Has this been verified with the Jacobian checker? I think the documentation on the KKS derivation page should be updated a little, since that page mentions c_a specifically and it might be confusing now that c_a no longer is checked specficially in the Jacobian. Maybe just mention that the second derivative with respect to c_a is captured by d2Fadcadarg[cvar].

Also, for my own understanding, was the separate check for `(jvar == _ca_var)` always unnecessary, or have the new Jvar Map interface functions that you mentioned enabled that check to be removed?

Reply:
The separate check was always unnecessary